commercial/app/model/commercial/jobs/JobsFeed.scala

```diff
@@ -1,52 +1,42 @@
 package model.commercial.jobs
 
+import java.lang.System.currentTimeMillis
+import java.util.concurrent.TimeUnit.MILLISECONDS
+
+import commercial.feeds.{MissingFeedException, ParsedFeed, SwitchOffException}
 import common.{ExecutionContexts, Logging}
-import conf.CommercialConfiguration
-import conf.switches.Switches.JobFeedSwitch
-import model.commercial.{FeedMissingConfigurationException, FeedReader, FeedRequest}
-import org.joda.time._
+import conf.Configuration.commercial.merchandisingFeedsRoot
+import conf.switches.Switches
+import services.S3
 
 import scala.concurrent.Future
-import scala.concurrent.duration._
+import scala.concurrent.duration.Duration
+import scala.util.control.NonFatal
 import scala.xml.{Elem, XML}
 
 object JobsFeed extends ExecutionContexts with Logging {
 
-  // url changes daily so cannot be val
-  def maybeUrl = {
-
-    /*
-     * Using offset time because this appears to be how the URL is constructed.
-     * With UTC time we lose the feed for 2 hours at midnight every day.
-     */
-    val feedDate = new DateTime(DateTimeZone.forOffsetHours(-2)).toString("yyyy-MM-dd")
-
-    val urlTemplate = CommercialConfiguration.getProperty("jobs.api.url.template")
-    urlTemplate map (_ replace("yyyy-MM-dd", feedDate))
-  }
-
   def parse(xml: Elem): Seq[Job] = for {
     jobXml <- xml \\ "Job"
     if (jobXml \ "RecruiterLogoURL").nonEmpty
     if (jobXml \ "RecruiterName").text != "THE GUARDIAN MASTERCLASSES"
   } yield Job(jobXml)
 
-  def loadAds(): Future[Seq[Job]] = {
-    maybeUrl map { url =>
-    val request = FeedRequest(
-      feedName = "Jobs",
-      switch = JobFeedSwitch,
-      url,
-      responseEncoding = Some("utf-8"),
-      timeout = 30.seconds
-    )
-    FeedReader.readSeq[Job](request) { body =>
-      parse(XML.loadString(body.dropWhile(_ != '<')))
-    }
-    } getOrElse {
-      log.warn(s"Missing URL for Jobs feed")
-      Future.failed(FeedMissingConfigurationException("Jobs"))
+  def parsedJobs(feedName: String): Future[ParsedFeed[Job]] = {
+    Switches.JobFeedSwitch.isGuaranteedSwitchedOn flatMap { switchedOn =>
+      if (switchedOn) {
+        val start = currentTimeMillis
+        S3.get(s"$merchandisingFeedsRoot/$feedName") map { body =>
```

Comment:
Should wrap this in a very obvious function that this is the "persistent" object that is giving us the safety.